src/Memory_Map.md

```diff
@@ -5,18 +5,92 @@ The Game Boy has a 16-bit address bus, which is used to address ROM, RAM, and I/
 A000        | BFFF      | 8 KiB External RAM                                               | From cartridge, switchable bank if any
 C000        | CFFF      | 4 KiB Work RAM (WRAM)                                            |
-D000        | DFFF      | 4 KiB Work RAM (WRAM)                                            | In CGB mode, switchable bank 1\~7
-E000        | FDFF      | Mirror of C000\~DDFF (ECHO RAM)                                  | Nintendo says use of this area is prohibited.
+D000        | DFFF      | 4 KiB Work RAM (WRAM)                                            | In CGB mode, switchable bank 1–7
+E000        | FDFF      | [Echo RAM](<#Echo RAM>) (mirror of C000–DDFF)                    | Nintendo says use of this area is prohibited.
 FE00        | FE9F      | [Object attribute memory (OAM)](<#Object Attribute Memory (OAM)>) |
-FEA0        | FEFF      | Not Usable                                                       | Nintendo says use of this area is prohibited
+FEA0        | FEFF      | [Not Usable](<#FEA0–FEFF range>)                                 | Nintendo says use of this area is prohibited.
 FF00        | FF7F      | [I/O Registers](<#I/O Ranges>)                                   |
 FF80        | FFFE      | High RAM (HRAM)                                                  |
 FFFF        | FFFF      | [Interrupt](#Interrupts) Enable register (IE)                    |
 
+## I/O Ranges
+
+The Game Boy uses the following I/O ranges:
+
+Start   | End     | First appeared | Purpose
+--------|---------|----------------|-------------
+$FF00   |         |       DMG      | [Joypad input](<#FF00 — P1/JOYP: Joypad>)
+$FF01   |  $FF02  |       DMG      | [Serial transfer](<#Serial Data Transfer (Link Cable)>)
+$FF04   |  $FF07  |       DMG      | [Timer and divider](<#Timer and Divider Registers>)
+$FF10   |  $FF26  |       DMG      | [Audio](<#Audio Registers>)
+$FF30   |  $FF3F  |       DMG      | [Wave pattern](<#FF30–FF3F — Wave pattern RAM>)
+$FF40   |  $FF4B  |       DMG      | LCD [Control](<#FF40 — LCDC: LCD control>), [Status](<#FF41 — STAT: LCD status>), [Position, Scrolling](<#LCD Position and Scrolling>), and [Palettes](<#Palettes>)
+$FF4F   |         |       CGB      | [VRAM Bank Select](<#FF4F — VBK (CGB Mode only): VRAM bank>)
+$FF50   |         |       DMG      | Set to non-zero to disable boot ROM
+$FF51   |  $FF55  |       CGB      | [VRAM DMA](<#LCD VRAM DMA Transfers>)
+$FF68   |  $FF6B  |       CGB      | [BG / OBJ Palettes](<#LCD Color Palettes (CGB only)>)
+$FF70   |         |       CGB      | [WRAM Bank Select](<#FF70 — SVBK (CGB Mode only): WRAM bank>)
+
+## VRAM memory map
+
+VRAM is, by itself, normal RAM, and may be used as such; however, the PPU interprets it in specific ways.
+
+Bank 1 does not exist except on CGB, where it can be switched to (only in CGB Mode) using [the `VBK` register](<#FF4F — VBK (CGB Mode only): VRAM bank>).
```

Comment:
Remove repetition:

```suggestion
Bank 1 does not exist except on CGB, where it can be switched to using [the `VBK` register](<#FF4F — VBK (CGB Mode only): VRAM bank>).
```

Reply:
I don't think that's repetition, as this wording is specifically designed to single out CGB in compatibility mode; where the bank exists in hardware, but the register is disabled so the bank can't be accessed.